### PR TITLE
OSDOCS-11208 OLM, builds, & etcd bugs 4.17 RN

### PR DESCRIPTION
**Pull request by cbippley**
Version(s): 4.17

Issue: [OSDOCS-11208](https://issues.redhat.com/browse/OSDOCS-11208)

Link to [docs preview](https://82494--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-17-release-notes.html#ocp-4-17-bug-fixes_release-notes)

QE review: Not needed
- [ ] QE has approved this change.

Additional information:

### CODE REVIEW

File: release_notes/ocp-4-17-release-notes.adoc [hunk 1]
```diff
@@ -1313,6 +1317,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
 [id="ocp-4-17-olm-bug-fixes_{context}"]
 ==== Operator Lifecycle Manager (OLM)
 
+* Previously, clusters with many custom resources (CRs) experienced timeouts from the API server and stranded updates where the only workaround was to uninstall and then reinstall the stranded Operators. This occurred because OLM evaluated potential updates by using a dynamic client lister. With this fix, OLM uses a paging lister for custom resource definitions (CRDs) to avoid timeouts and stranded updates. (link:https://issues.redhat.com/browse/OCPBUGS-41549[*OCPBUGS-47549*])
+
+* Previously, catalog source pods could not recover from a cluster node failure when the `registryPoll` was unset. With this fix, OLM updates its logic for checking for dead pods, and as a result, catalog source pods now recover from node failures as expected. (link:https://issues.redhat.com/browse/OCPBUGS-39574[*OCPBUGS-39574*])
```
Comment (jldohmann) [inline release_notes/ocp-4-17-release-notes.adoc:1322]: anything in backticks should be followed by what it is, similar to how it's done in the [quick markup reference table](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#quick-markup-reference). the separate sentences at the end is just a suggestion because it was a little long:
```suggestion
* Previously, catalog source pods could not recover from a cluster node failure when the `registryPoll` parameter was unset. With this fix, OLM updates its logic for checking for dead pods. As a result, catalog source pods now recover from node failures as expected. (link:https://issues.redhat.com/browse/OCPBUGS-39574[*OCPBUGS-39574*])
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 2]
```diff
@@ -1313,6 +1317,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
...
+
+* Previously, if you tried to install a previously deleted Operator after a {product-title} update, the installation might fail. This occured because OLM could not find previously created bundle unpack jobs. With this fix, OLM correctly installs previously installed Operators. (link:https://issues.redhat.com/browse/OCPBUGS-32439[*OCPBUGS-32439*])
```
Comment (jldohmann) [inline release_notes/ocp-4-17-release-notes.adoc:1324]:
```suggestion
* Previously, if you tried to install a previously deleted Operator after a {product-title} update, the installation might fail. This occurred because OLM could not find previously created bundle unpack jobs. With this fix, OLM correctly installs previously installed Operators. (link:https://issues.redhat.com/browse/OCPBUGS-32439[*OCPBUGS-32439*])
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 3]
```diff
@@ -1313,6 +1317,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
...
+
+* Previously, when a new version of a custom resource definition (CRD) specified a new conversion strategy, this conversion strategy was expected to successfully convert resources. However, OLM cannot run the new conversion strategies for CRD validation without actually performing the update operation. With this release, OLM generates a warning message during the update process when CRD validations fail with the existing conversion strategy, and the new conversion strategy is specified in the new version of the CRD. (link:https://issues.redhat.com/browse/OCPBUGS-31522[*OCPBUGS-31522*])
+
+* If the `spec.grpcPodConfig.securityContextConfig` field is unset in `CatalogSource` objects in namespaces with `PodSecurityAdmission` (PSA) level `restricted` enforced, the OLM Catalog Operator now configures the catalog pod with the `securityContexts` necessary to pass PSA validation. (link:https://issues.redhat.com/browse/OCPBUGS-29729[*OCPBUGS-29729*])
```
Comment (jldohmann) [inline release_notes/ocp-4-17-release-notes.adoc:1328]: this first sentence is really hard to parse:
```suggestion
* If the `spec.grpcPodConfig.securityContextConfig` field is unset in `CatalogSource` objects in namespaces with a `PodSecurityAdmission` (PSA) level value of `restricted`, the OLM Catalog Operator now configures the catalog pod with the `securityContexts` necessary to pass PSA validation. (link:https://issues.redhat.com/browse/OCPBUGS-29729[*OCPBUGS-29729*])
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 4]
```diff
@@ -1313,6 +1317,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
...
+
+* Previously, the `catalogd-controller-manager` pod could become unschedulable, and the OLM Operator would fail to install. With this fix, CPU requests are reduced for the related resources, and the issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-29705[*OCPBUGS-29705*])
```
Comment (jldohmann) [inline release_notes/ocp-4-17-release-notes.adoc:1330]: i tried looking up the word "unschedulable" and i don't think it's a real word EXCEPT in the context of containers 😭 ...my suggestion here is to just spell out the whole meaning with actual words:
```suggestion
* Previously, the `catalogd-controller-manager` pod might not be deployed to a node despite being in the scheduling queue, and the OLM Operator would fail to install. With this fix, CPU requests are reduced for the related resources, and the issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-29705[*OCPBUGS-29705*])
```

**Review by jldohmann — COMMENTED** [review]
it's looking good! some comments below

File: release_notes/ocp-4-17-release-notes.adoc (hunk 3 above)
Comment (cbippley) [reply release_notes/ocp-4-17-release-notes.adoc:1328]: I cleaned it up the best I could.

File: release_notes/ocp-4-17-release-notes.adoc [hunk 5]
```diff
@@ -1264,6 +1266,8 @@ OpenShift SDN network plugin was deprecated in 4.15 and 4.16. With this release,
 [id="ocp-4-17-cloud-etcd-operator-bug-fixes_{context}"]
 ==== etcd Cluster Operator
 
+* Previous versions of the etcd Operator checked the health of etcd members in serial with an all-member timeout that matched the single-member timeout. As a result, one slow member check could consume the entire timeout and cause later member checks to fail, regardless of the health of that later member. In this release, the etcd Operator checks the health of members in parallel, so the health and speed of one member's check doesn't affect the other members' checks. (link:https://issues.redhat.com/browse/OCPBUGS-36301[*OCPBUGS-36301*])
```
Comment (skopacz1) [inline release_notes/ocp-4-17-release-notes.adoc:1269]:
```suggestion
* Previous versions of the etcd Operator checked the health of etcd members in serial with an all-member timeout that matched the single-member timeout. As a result, one slow member check could consume the entire timeout and cause later member checks to fail, regardless of the health of that later member. In this release, the etcd Operator checks the health of members in parallel, so the health and speed of one member's check does not affect the other members' checks. (link:https://issues.redhat.com/browse/OCPBUGS-36301[*OCPBUGS-36301*])
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 6]
```diff
@@ -1336,6 +1340,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
...
 
+* Previously, clusters with many custom resources (CRs) experienced timeouts from the API server and stranded updates where the only workaround was to uninstall and then reinstall the stranded Operators. This occurred because OLM evaluated potential updates by using a dynamic client lister. With this fix, OLM uses a paging lister for custom resource definitions (CRDs) to avoid timeouts and stranded updates. (link:https://issues.redhat.com/browse/OCPBUGS-41549[*OCPBUGS-47549*])
+
+* Previously, catalog source pods could not recover from a cluster node failure when the `registryPoll` parameter was unset. With this fix, OLM updates its logic for checking for dead pods. As a result, catalog source pods now recover from node failures as expected. (link:https://issues.redhat.com/browse/OCPBUGS-39574[*OCPBUGS-39574*])
+
+* Previously, if you tried to install a previously deleted Operator after a {product-title} update, the installation might fail. This occurred because OLM could not find previously created bundle unpack jobs. With this fix, OLM correctly installs previously installed Operators. (link:https://issues.redhat.com/browse/OCPBUGS-32439[*OCPBUGS-32439*])
```
Comment (skopacz1) [inline release_notes/ocp-4-17-release-notes.adoc:1347]: Article plus a hyphen to form a compound adjective:
```suggestion
* Previously, if you tried to install a previously-deleted Operator after an {product-title} update, the installation might fail. This occurred because OLM could not find previously created bundle unpack jobs. With this fix, OLM correctly installs previously installed Operators. (link:https://issues.redhat.com/browse/OCPBUGS-32439[*OCPBUGS-32439*])
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 7]
```diff
@@ -1336,6 +1340,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
...
+
+* Previously, if the `spec.grpcPodConfig.securityContextConfig` field in `CatalogSource` objects was unset within namespaces with a `PodSecurityAdmission` (PSA) level value of `restricted`, the catalog pod would not pass PSA validation. With this release, the OLM Catalog Operator now configures the catalog pod with the `securityContexts` necessary to pass PSA validation.(link:https://issues.redhat.com/browse/OCPBUGS-29729[*OCPBUGS-29729*])
+
+* Previously, the `catalogd-controller-manager` pod might not be deployed to a node despite being in the scheduling queue, and the OLM Operator would fail to install. With this fix, CPU requests are reduced for the related resources, and the issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-29705[*OCPBUGS-29705*])
```
Comment (skopacz1) [inline release_notes/ocp-4-17-release-notes.adoc:1353]: Should this be in past tense here?
```suggestion
* Previously, the `catalogd-controller-manager` pod might not have been deployed to a node despite being in the scheduling queue, and the OLM Operator would fail to install. With this fix, CPU requests are reduced for the related resources, and the issue no longer occurs. (link:https://issues.redhat.com/browse/OCPBUGS-29705[*OCPBUGS-29705*])
```

File: release_notes/ocp-4-17-release-notes.adoc [hunk 8]
```diff
@@ -1336,6 +1340,20 @@ link:https://issues.redhat.com/browse/OCPBUGS-39428[(*OCPBUGS-39428*)]
...
+
+* Previously, sometimes the Catalog Operator attempted to connect to deleted catalog sources that were stored in the cache. With this fix, the Catalog Operator queries a client to list the catalog sources on a cluster. (link:https://issues.redhat.com/browse/OCPBUGS-8659[*OCPBUGS-8659*])
```
Comment (skopacz1) [inline release_notes/ocp-4-17-release-notes.adoc:1355]: Suggesting to move "sometimes" since "Previously, sometimes" sounds a little awkward imo
```suggestion
* Previously, the Catalog Operator sometimes attempted to connect to deleted catalog sources that were stored in the cache. With this fix, the Catalog Operator queries a client to list the catalog sources on a cluster. (link:https://issues.redhat.com/browse/OCPBUGS-8659[*OCPBUGS-8659*])
```

**Review by skopacz1 — COMMENTED** [review]
A few small nits, otherwise LGTM! Let me know when they're addressed and I can merge

**Comment by skopacz1** [comment]
LGTM, thanks
